eZ/Publish/Core/Persistence/Cache/UserHandler.php

```diff
@@ -339,4 +349,21 @@ public function unAssignRole($contentId, $roleId)
 
         return $return;
     }
+
+    /**
+     * @see eZ\Publish\SPI\Persistence\User\Handler::removeRoleAssignment
+     */
+    public function removeRoleAssignment($roleAssignmentId)
+    {
+        $this->logger->logCall(__METHOD__, array('assignment' => $roleAssignmentId));
+        $return = $this->persistenceHandler->userHandler()->removeRoleAssignment($roleAssignmentId);
+
+        // We don't know the contentId, so clear all assignment cache.
+        // TODO: fetch contentId to avoid this?
```

Comment:
I confirm.

Status is that we either need to find a way to move this cache to api in the future where we have more info because they take objects as arguments *(this will need lots of refactoring to separate Business Logic from Permission logic, as we don't want to cache per user)*, and/or refactor SPI to be modeled like API: Take objects as arguments instead of just ID's for operations like delete/move/...  *Last option might make sense given we for paranoia reasons tends to reload SPI objects inAPI BL anyway, so we tend to have the object there. And actually the options are not mutually exclusive, as both changes might make sense over time.*

@glye just add `TIMBER` comment anywhere you wipe out all cache, or all content cache or similar, we will need to deal with these very soon somehow, as they limit scalability on active installs, so god to easily find them.